### PR TITLE
Adds a list-checkpoints command

### PR DESCRIPTION
**Pull request by hmoniz**
Adds a command that lists a range of checkpoints. This is predicated on a Lotus JSON-RPC method for getting a range of checkpoints.

### CODE REVIEW

File: src/cli/commands/manager/list_checkpoints.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        let json_rpc_client = JsonRpcClientImpl::new(url, None);
+
+        let params = ListCheckpointsParams {
+            subnet_id: arguments.subnet_id.clone(),
+            from_epoch: arguments.from_epoch,
+            to_epoch: arguments.to_epoch,
+        };
+
+        let checkpoints = json_rpc_client
+            .request::<HashMap<SubnetID, SubnetInfo>>(
```
Comment (cryptoAtwill) [inline src/cli/commands/manager/list_checkpoints.rs:39]: This `SubnetInfo` should be `Checkpoint`?

**Review by cryptoAtwill — CHANGES_REQUESTED** [review]

File: src/cli/commands/manager/list_checkpoints.rs (hunk 1 above)
Comment (hmoniz) [reply src/cli/commands/manager/list_checkpoints.rs]: Yes, fixed

File: src/cli/commands/manager/list_checkpoints.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        let json_rpc_client = JsonRpcClientImpl::new(url, None);
+
+        let params = ListCheckpointsParams {
+            subnet_id: arguments.subnet_id.clone(),
+            from_epoch: arguments.from_epoch,
+            to_epoch: arguments.to_epoch,
+        };
+
+        let checkpoints = json_rpc_client
+            .request::<HashMap<SubnetID, Checkpoint>>(
```
Comment (adlrocha) [inline src/cli/commands/manager/list_checkpoints.rs:39]: Typo? The key of the hashmap should be the epoch, right? 
```suggestion
            .request::<HashMap<Epoch, Checkpoint>>(
```

**Review by adlrocha — CHANGES_REQUESTED** [review]
It was a bit hard to review, I think it picked up many changes from the base branch that you used for some reason (which have already merged to main). 

Overall, LGTM, but for it to work you'll need to incorporate the changes from this PR: https://github.com/consensus-shipyard/ipc-agent/pull/136. The current way we are using to deserialize the list checkpoints wouldn't work, unfortunately.

File: src/lotus/client.rs [hunk 3]
```diff
@@ -337,6 +399,32 @@ impl<T: JsonRpcClient + Send + Sync> LotusClient for LotusJsonRPCClient<T> {
...
+            {
+                "Parent": parent,
+                "Actor": actor
+            },
+            from_epoch,
+            to_epoch
+        ]);
+        let r = self
+            .client
+            .request::<Vec<Checkpoint>>(methods::IPC_LIST_CHECKPOINTS, params)
```
[the remark this answers is not in the document]
Comment (hmoniz) [reply src/lotus/client.rs]: Fixed

File: src/cli/commands/manager/list_checkpoints.rs (hunk 2 above)
Comment (hmoniz) [reply src/cli/commands/manager/list_checkpoints.rs]: Fixed. This actually should be Vec<Checkpoint>

**Review by adlrocha — APPROVED** [review]
Tested and fixed sync

**Pull request merged**